### PR TITLE
Implement `Base.show` to remove the errors in the REPL

### LINKED ISSUE #7: Printable PSDMatrix
Currently, when printing a PSDMatrix, one gets an error:

```
Error showing value of type PSDMatrix{Float64, LowerTriangular{Float64, Adjoint{Float64, Matrix{Float64}}}}:
ERROR: getindex not defined for PSDMatrix{Float64, LowerTriangular{Float64, Adjoint{Float64, Matrix{Float64}}}}
Stacktrace:
```
Would be great to have this fixed.

### PR DESCRIPTION
**Pull request by nathanaelbosch**
Fixes #7

### CODE REVIEW

File: src/PSDMatrices.jl [hunk 1]
```diff
@@ -18,6 +18,16 @@ inv(M::PSDMatrix) = PSDMatrix(inv(M.R'))
 /(B::AbstractVecOrMat, A::PSDMatrix) = B / A.R / A.R'
 copy(M::PSDMatrix) = PSDMatrix(copy(M.R))
 ==(M1::PSDMatrix, M2::PSDMatrix) = M1.R == M2.R  # todo: same as isequal()?!
+# Base.show(io::IO, x::MyString) = print(io, x.s)
```
Comment (pnkraemer) [inline src/PSDMatrices.jl]:
```suggestion
```

File: src/PSDMatrices.jl [hunk 2]
```diff
@@ -18,6 +18,16 @@ inv(M::PSDMatrix) = PSDMatrix(inv(M.R'))
...
+show(io::IO, M::PSDMatrix) = begin
+    print(io, "$(size(M,1))x$(size(M,2)) $(typeof(M)); R=")
+    show(io, M.R)
+end
+show(io::IO, m::MIME"text/plain", M::PSDMatrix) = begin
+    println(io, "$(size(M,1))x$(size(M,2)) $(typeof(M)) ")
+    print(io, " Right square root: ")
+    show(io, m, M.R)
+end
```
Comment (pnkraemer) [inline src/PSDMatrices.jl]: I dont understand the need for the second function, but I trust you here :) 

Maybe for consistency with the print message above, add the "R="? But feel free to ignore :)
```suggestion
show(io::IO, M::PSDMatrix) = begin
    print(io, "$(size(M,1))x$(size(M,2)) $(typeof(M)); R=")
    show(io, M.R)
end
show(io::IO, m::MIME"text/plain", M::PSDMatrix) = begin
    println(io, "$(size(M,1))x$(size(M,2)) $(typeof(M)) ")
    print(io, " Right square root: R=")
    show(io, m, M.R)
end
```

**Review by pnkraemer — APPROVED** [review]
Thanks! I think there is one dead line left in the code, but other than that, you have my blessing to merge :)

File: src/PSDMatrices.jl (hunk 2 above)
Comment (pnkraemer) [reply src/PSDMatrices.jl]: (I am also not sure how standard the show() = begin ... end syntax is, but it seems to work well and I am not the correct person to complain about standard syntax :P)

Comment (pnkraemer) [reply src/PSDMatrices.jl]: (I am also not sure how standard the show() = begin ... end syntax is, but it seems to work well and I am not the correct person to complain about standard syntax :P)

Comment (nathanaelbosch) [reply src/PSDMatrices.jl]: I changed the function definitions to be non-anonymous :)

**Review by pnkraemer — APPROVED** [review]

**Pull request merged**